File: src/main/java/net/sf/rails/game/RailsAbstractItem.java
package net.sf.rails.game;

import net.sf.rails.game.state.AbstractItem;

/**
 * RailsAbstractItem adds Rails specific methods to AbstractItem
 */
public abstract class RailsAbstractItem extends AbstractItem implements RailsItem {

    protected RailsAbstractItem(RailsItem parent, String id) {
        super(parent, id);
    }
    
    @Override
    public RailsItem getParent() {
        return (RailsItem)super.getParent();
    }
    
    @Override
    public RailsRoot getRoot() {
        return (RailsRoot)super.getRoot();
    }
    
}

File: dotnet/RailsLib.Net/Net/Game/RailsAbstractItem.cs
using GameLib.Net.Game.State;
using System;


namespace GameLib.Net.Game
{
    abstract public class RailsAbstractItem : AbstractItem, IRailsItem
    {
        protected RailsAbstractItem(IRailsItem parent, string id) : base(parent, id)
        {
        }

        new public IRailsItem Parent
        {
            get
            {
                return (IRailsItem)base.Parent;
            }
        }

        //public IItem IItem.Parent => Parent;

        new public RailsRoot GetRoot
        {
            get
            {
                return (RailsRoot)base.GetRoot;
            }
        }
    }
}
